src/gba/audio.c

```diff
@@ -57,6 +57,7 @@ void GBAAudioInit(struct GBAAudio* audio, size_t samples) {
 
 void GBAAudioReset(struct GBAAudio* audio) {
 	GBAudioReset(&audio->psg);
+	memset(audio->psg.ch3.wavedata8, 0, sizeof(audio->psg.ch3.wavedata8));
```

Comment:
This should probably be moved to `GBAudioReset` and the pattern initialization be moved to `GBSkipBIOS` then.